src/osd/ClassHandler.cc

```diff
@@ -75,9 +83,11 @@ int ClassHandler::open_all_classes()
 
 void ClassHandler::shutdown()
 {
-  for (auto& cls : classes) {
-    if (cls.second.handle) {
-      dlclose(cls.second.handle);
+  if (!disable_dlclose) {
```

Comment:
i am not following you. as i think you are not addressing my comment. also, could you elaborate a little bit? whose destructor?